R/torchaudio-package.R

```diff
@@ -1,15 +1,8 @@
-## usethis namespace: start
-#' @importFrom Rcpp sourceCpp
-#' @useDynLib torchaudio, .registration = TRUE
-## usethis namespace: end
-NULL
-
 utils::globalVariables(c("..", "N"))
 
-
 .onLoad <- function(libname, pkgname) {
   op <- options(
```

Comment:
I think we should not have a `options` call in `.onLoad`. This will override the user's setting when loading the package. Instead, we should use `getOption("opt", default = <the default>) `.

Reply:
thanks @dfalbel ! Does what I am doing now address your two loading-related comments? 

I now have two different options, one to be employed by the user, "torchaudio.loader", and one to be used internally, "torchaudio_backend". For the latter, I now have a getter and a setter, which both work with strings, not functions. 

The drawback, if it is one, is that now `torchaudio_load` converts the string to a function. That's not really on purpose, but comes from the fact that I can't do `rlang::as_function` in, or from, `.onload`. Do you have a better idea or do you think it is ok as-is? (I mean, impact on performance can't be that big, I guess.)

Reply:
hm, on CI the tests fail with the latest commit, so anyway there must be a problem ... locally tests work fine ...

Reply:
OK I see, the problem is how I'm running tests locally, namely, after loading ... and some tests use the function, not the string, to set the backend ... So I could change that, but maybe I'll wait to see what you say, because I might have to change that back :-)

Reply:
IMHO you should have a single option `torchaudio.backend`. You can then have 2 functions: `get_torchaudio_backend()` that just does `getOptionn("torchaudio.backend", default="av")` and `get_torchaudio_loader()` which calls `get_torch_audio_backend()` and decides which 'loader' it wants to use. Ideally, you shouldn't set any option in `.onLoad()`.

Reply:
Oh, I think I understand now! I somehow didn't get that I should not do _anything_ on-load ... actually this means I don't even need the file anymore, then ...
Let me see how this works :-)

I know this is unrelated to this discussion, but do you know anything about https://github.com/mlverse/torchaudio/blob/main/R/temp.R? I could either remove it, or put it into some `utils.R` if it makes sense to keep around.

Reply:
Oh, another question. I made get_`torchaudio_backend() `and `get_torchaudio_backend() `private, so the user only works via setting the option. That would still make sense with your above suggestion, right?

Reply:
So sorry, still not really sure this is what you mean ... I now have

```
#' @keywords internal
set_audio_backend <- function(backend) {
  options("torchaudio.backend" = backend)
}

#'  @keywords internal
get_audio_backend <- function() {
  rlang::as_function(getOption("torchaudio.backend", default="av_loader"))
}
```

... and NULL in `.onload`.

Reply:
This looks like a help script to convert from python functions to R. You can move it in a `tools` folder in the root directory.
Yes, it make sense to make it private only.

Reply:
@dfalbel thanks, will do that and keep it, can always turn out to be useful!